client/src/js/services/PrototypeApiService.js

```diff
@@ -244,17 +244,32 @@ function PrototypeApiService($http, util) {
 
     // append the id to the base url
     var target = this.url.concat('reports/', param);
+    return reportBuilder(this, param, filetype);
+  }
+
+  /**
+   * Receipt
+   */
+  function receipt(param, filetype) {
+
+    // append the id to the base url
+    var target = this.url.concat('receipts/', param);
```

Comment:
Putting this function in `PrototypeApiService` is a really interesting idea.  It means that it naturally defines APIs for reports/receipts as follows:

```js
var voucherService = new Api('vouchers/');
voucherService.receipt(uuid); // sends GET vouchers/receipts/uuid

var accountService = new Api('accounts/');
accountService.receipt(uuid); // sends GET accounts/receipts/uuid

var inventoryService = new Api('inventory/');
inventoryService.receipt(uuid); // sends GET inventory/receipts/uuid
```
That's a pretty cool API standard.  I like it.

@sfount, what do you think of this standard for APIs?  Should receipts be part of each API, or a separate API altogether?  You could also imagine...
```
GET /receipts/vouchers/:uuid
GET /receipts/accounts/:id
GET /receipts/inventory/:id
```

Which do you prefer?